source/utils.go

```diff
@@ -0,0 +1,62 @@
+/*
+Copyright 2025 The Kubernetes Authors.
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+    http://www.apache.org/licenses/LICENSE-2.0
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package source
+
+import (
+	"fmt"
+	"net/netip"
+	"strings"
+
+	"sigs.k8s.io/external-dns/endpoint"
+)
+
+// suitableType returns the DNS resource record type suitable for the target.
+// In this case type A/AAAA for IPs and type CNAME for everything else.
+func suitableType(target string) string {
+	netIP, err := netip.ParseAddr(target)
+	if err == nil && netIP.Is4() {
```

Comment:
We should always test for err != nil first:

```
if err != nil {
  return cname 
} else {
  switch {
    case ipv4:
...
  }
}

return cname 
```

Reply:
done